src/istgt_iscsi.c

```diff
@@ -7180,6 +7208,41 @@ istgt_iscsi_drop_all_conns(CONN_Ptr conn)
 	return (0);
 }
 
+#ifdef REPLICATION
+// istgt_iscsi_get_existing_live_conn returns existing live
+// connection other than current connection
+// Note: This function should be called by taking lock on g_conns_mutex
+CONN_Ptr
+istgt_iscsi_get_existing_live_conn(CONN_Ptr conn)
+{
+	CONN_Ptr existing_conn = NULL;
+	int i;
+
+	ASSERT(MTX_LOCKED(&g_conns_mutex));
+	for (i = 0; i <= g_max_connidx; i++) {
+		existing_conn = g_conns[i];
+		if (existing_conn == NULL)
+			continue;
+		if (existing_conn == conn)
+			continue;
+		// If new connection is totally different from existing connection
+		// then check the status of existing connection if it is active
+		// return true
+		// Note:
+		//	- Here we are assuming initiator will always use same address for
+		// 	  to interact with iSCSI target
+		//	- Initiator name will get populated only after login request so we
```

Comment:
is it not possible to check this with iqn name? I remember we discussed to move it at login time?

Reply:
There exist a race between two different logins... To avoid race we need to add a lot of changes to handle it easily moved before logging time... Since the check is protected by cStor macro I thought we are good.